tools/cli/helpers/doc-parser/src/class-doc-parser.php

```diff
@@ -104,15 +146,304 @@ function ( $item ) use ( $ignore ) {
+				$tags[]      = new ParamTagValueNode( $type, true, $name, $description, false );
+			}
+		}
+
+		return $tags;
+	}
+
+	/**
+	 * Returns a list of PHP files in a folder, recursing into subfolders. Heavily inspired by
+	 * the WordPress PHPDoc parser.
+	 *
+	 * @see https://github.com/WordPress/phpdoc-parser/blob/7fc2227d2d4fb73f9f0b6e233413f3f9f9840e80/lib/runner.php#L17
+	 *
+	 * @param string $directory the folder to look in.
+	 *
+	 * @return array an array of filenames.
+	 * @throws \Exception $e If unable to traverse the filesystem.
+	 */
+	public function get_wp_files( $directory ) {
+		$iterableFiles = new \RecursiveIteratorIterator(
+			new \RecursiveDirectoryIterator( $directory )
+		);
+		$files         = array();
+
+		try {
+			foreach ( $iterableFiles as $file ) {
+				if ( 'php' !== $file->getExtension() ) {
+					continue;
+				}
+
+				$files[] = $file->getPathname();
+			}
+		} catch ( \UnexpectedValueException $exc ) {
+			throw new \Exception( sprintf( 'Directory [%s] contained a directory we can not recurse into', $directory ) );
+		}
+
+		return $files;
+	}
+
+	/**
+	 * Returns an array of docblock annotations for apply_filter function calls, with keys being names of filters
+	 * used.
+	 *
+	 * @param array $nodes  Parser node objects for hook calls.
+	 * @return array docblock annotations.
+	 */
+	public function get_filter_calls( $nodes ): array {
+
+		$blocks = array();
+
+		foreach ( $nodes as $node ) {
+
+			$arguments = $node->getArgs();
+			$hook_name = array_shift( $arguments );
+
+			// Purging any comments that could have been attributed to this argument.
+			$comments = $hook_name->getComments();
+			if ( ! empty( $comments ) ) {
+				$hook_name->setAttribute( 'comments', null );
+			}
+
+			$new_block = array(
+				'type'      => $node->name->name === 'apply_filters' ? 'filter' : 'action',
+				'line'      => $node->getLine(),
+				'end_line'  => $node->getEndLine() > 0 ? $node->getEndLine() : $node->getLine(),
+				'name'      => $this->pretty_print_hook_name( $hook_name ),
+				'arguments' => array(),
+			);
+
+			foreach ( $arguments as $argument ) {
+				$new_block['arguments'][] = $this->printer->prettyPrint( array( $argument ) );
+			}
+			$blocks[] = $new_block;
+		}
+
+		return $blocks;
+	}
+
+	/**
+	 * Pretty prints the name for the hook, taking an argument object as input.
+	 *
+	 * @suppress PhanRedundantConditionInLoop Phan is having trouble getting some of nikic/php-parser type definitions.
+	 *
+	 * @param Node\Arg $argument the first argument to the apply_filter or do_action call.
+	 * @return String pretty printed argument name.
+	 * @throws \Exception On an unexpected argument component.
+	 */
+	public function pretty_print_hook_name( Node\Arg $argument ): string {
+
+		if (
+			$argument->value instanceof Node\Scalar\String_
+				|| $argument->value instanceof Node\Expr\ConstFetch
+				|| $argument->value instanceof Node\Expr\ClassConstFetch
```

Comment:
These cases probably aren't too useful as-is for the documentation. We want the doc to say "stats_array" or "earn_premium_content_subscription_service", not "self::STATS_ARRAY_TO_STRING_FILTER" or "PAYWALL_FILTER".

Looks like those are the only two we currently have in jetpack-production, BTW. In both cases the constant is defined in the same file, so (probably in a followup) we could probably have some code check for definitions.

Anyway, leaving all that for a followup seems fine to me.